creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1146,18 +1190,33 @@ class SnapbackSM {
     }
   }
 
-  _addToStateMachineQueueDecisionTree(decisionTree, stage, data = {}) {
+  _addToStateMachineQueueDecisionTree(
+    decisionTree,
+    jobId,
+    stage,
+    data = {},
+    log = true
+  ) {
     const obj = { stage, data, time: Date.now() }
 
+    let logStr = `processStateMachineOperation() ${jobId} ${stage} - Data ${JSON.stringify(
+      data
+    )}`
+
     if (decisionTree.length > 0) {
       // Set duration if both objs have time field
       const lastObj = decisionTree[decisionTree.length - 1]
       if (lastObj && lastObj.time) {
         const duration = obj.time - lastObj.time
         obj.duration = duration
+        logStr += ` - Duration ${duration}ms`
       }
     }
     decisionTree.push(obj)
+
+    if (log) {
+      this.log(logStr)
+    }
```

Comment:
was this intentionally duplicating `_printStateMachineQueueDecisionTree` below?
esp with log defaulted true could get p noisy. maybe thats what you want tho, again your call - do what you think is best

Reply:
my thought here is that it'll be noisier but we'll be able to see what the last step that ran was before failing. otherwise we only know if the full task completed or not

Reply:
makes sense

Reply:
i meant more calling this function vs calling `printSMDecisionTree()` - the diff is that'll print full tree each time this'll just do current stage

this is all nitty haha - do as you please